cmd/agent/main.go

```diff
@@ -48,7 +48,7 @@ func main() {
 
 		if err == nil {
 			agent, err := app.NewAgent(ctx, tlsConfig)
-			if err == nil {
+			if err == nil && agent != nil {
```

Comment:
agent should never be nil if err is nil. I think we need to add an error to NewAgent.